packages/miniapp-utils/src/helpers/i18n.tsx

```diff
@@ -0,0 +1,479 @@
+    private readonly _loadDefaultMessages: () => Promise<MessagesShape>
+    private readonly _translations: Partial<Record<AvailableLocale, MessagesShape>> = {}
+    private _defaultMessages: MessagesShape | undefined
+    readonly localeCookieName: string = 'NEXT_LOCALE'
+
+    constructor (options: TranslationsHelperOptions<AvailableLocale, MessagesShape>) {
+        this._locales = new Set(options.locales)
+        this._defaultLocale = options.defaultLocale
+        this._loadMessages = options.loadMessages
+        this._loadDefaultMessages = options.loadDefaultMessages
+
+        if (options.localeCookieName) {
+            this.localeCookieName = options.localeCookieName
+        }
+
+        this.getTranslationsProvider = this.getTranslationsProvider.bind(this)
+        this.getUseTranslationsHook = this.getUseTranslationsHook.bind(this)
+        this.getUseTranslationsExtractorHook = this.getUseTranslationsExtractorHook.bind(this)
+        this.getPreferredLocale = this.getPreferredLocale.bind(this)
+        this.selectSupportedLocale = this.selectSupportedLocale.bind(this)
+        this.getTranslations = this.getTranslations.bind(this)
+        this.prefetchTranslations = this.prefetchTranslations.bind(this)
+    }
+
+    /**
+     * This util parses language-defining string according to RFC5646: https://datatracker.ietf.org/doc/html/rfc5646
+     * It also automatically detect and accept-language header format by enhancing result with quality info
+     */
+    static parseLocaleString (localeString: string): AcceptLanguageInfo {
+        const stringParts = localeString.trim().split(';')
+        const localeParts = stringParts[0].split('-')
+
+        const quality = stringParts.length > 1
+            ? parseFloat(stringParts[1].split('=')[1])
+            : 1.0
+
+        const locale: AcceptLanguageInfo = {
+            primary: localeParts[0],
+            extended: undefined,
+            script: undefined,
+            region: undefined,
+            quality,
+        }
+
+        let currentParser = 0
+        localePartsLoop: for (const localePart of localeParts) {
+            for (; currentParser < LOCALE_PARSING_OPTIONS.length; currentParser++) {
+                const { part, matcher } = LOCALE_PARSING_OPTIONS[currentParser]
+                if (matcher.test(localePart)) {
+                    locale[part] = localePart
+                    currentParser++
+                    continue localePartsLoop
+                }
+            }
+            break
+        }
+
+        return locale
+    }
+
+    /**
+     * Parses "Accept-Language" header value using "parseLocaleString" util and returns array of AcceptLanguageInfo
+     * sorted by descending quality.
+     *
+     * NOTE: Empty header or non-defined header is treated as "*"
+     */
+    static parseAcceptLanguageHeader (headerValue: Optional<string>): Array<AcceptLanguageInfo> {
```

Comment:
Why not default value?
```suggestion
    static parseAcceptLanguageHeader (headerValue: Optional<string> = '*'): Array<AcceptLanguageInfo> {
```